examples/sign-leaf-with-ca.rs

```diff
@@ -0,0 +1,99 @@
+use rcgen::{
+	date_time_ymd, BasicConstraints, Certificate, CertificateParams, CertificateSigningRequest,
+	DnType, ExtendedKeyUsagePurpose, IsCa, KeyUsagePurpose,
+};
+use time::{Duration, OffsetDateTime};
+use x509_parser::certification_request::X509CertificationRequest;
+use x509_parser::prelude::FromDer;
+
+/// Example demonstrating signing end-endity certificate with ca
+fn main() {
+	let ca = Ca::new();
+	let entity = Entity::new();
+
+	println!("directly signed end-entity certificate:");
+	let direct = entity
+		.certificate
+		.serialize_pem_with_signer(&ca.certificate)
+		.unwrap();
+	println!("{direct}");
+
+	println!("ca certificate:");
+	let pem = ca.certificate.serialize_pem().unwrap();
+
+	println!("{}", pem);
+}
+
+struct Ca {
+	certificate: Certificate,
+}
+
+impl Ca {
+	fn new() -> Self {
+		let mut params = CertificateParams::new(vec!["ca.some.host".to_owned()]);
+		let (yesterday, tomorrow) = y_t();
+		params.is_ca = IsCa::Ca(BasicConstraints::Unconstrained);
+		params.distinguished_name.push(DnType::CountryName, "BR");
+		params
+			.distinguished_name
+			.push(DnType::OrganizationName, "Crab widgits SE");
+		params.key_usages.push(KeyUsagePurpose::DigitalSignature);
+		params.key_usages.push(KeyUsagePurpose::KeyCertSign);
+		params.key_usages.push(KeyUsagePurpose::CrlSign);
+
+		params.not_before = yesterday;
+		params.not_after = tomorrow;
+		Self {
+			certificate: Certificate::from_params(params).unwrap(),
+		}
+	}
+
+	fn create_cert(&self, csr_pem: &str) -> String {
+		let csr_der = x509_parser::pem::parse_x509_pem(csr_pem.as_bytes())
+			.unwrap()
+			.1;
+		let csr = X509CertificationRequest::from_der(&csr_der.contents)
+			.unwrap()
+			.1;
+		csr.verify_signature().unwrap();
+		let csr = CertificateSigningRequest::from_der(&csr_der.contents).unwrap();
+		csr.serialize_pem_with_signer(&self.certificate).unwrap()
+	}
+}
+
+struct Entity {
+	certificate: Certificate,
+}
+
+impl Entity {
+	fn new() -> Self {
+		let name = "entity.other.host";
+		let mut params = CertificateParams::new(vec!["entity.other.host".to_owned()]);
+		let (yesterday, tomorrow) = y_t();
+		params.distinguished_name.push(DnType::CommonName, name);
+		params.use_authority_key_identifier_extension = true;
+		params
+			.subject_alt_names
+			.push(rcgen::SanType::DnsName(name.into()));
```

Comment:
Because you gave the same `nam` to the `CertificateParams` constructor this is actually adding a _second_ instance of the same domain name to the SAN extension. We'll want to avoid that. Probably deleting this explicit `subject_alt_names.push` is the best bet :+1: